### PR TITLE
Conversion to new style custom resource

### PR DESCRIPTION
**Pull request by axelrtgs**
### Description

Converts cookbook to new style custom resource

### Issues Resolved

Resolves undefined method issue

### Contribution Check List

- [x] All tests pass. 
- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by axelrtgs** [comment]
@tas50 Not sure how much i trust the tests but they pass and i was able to apply the ulimit i was expecting on my system. i didn't however use anything else other than user_ulimit resource in my local testing.

The travis build passed and the test recipe seems to include the different types so i assume its all good. I am certain some factoring out of useless lines of code may also be needed but i have no idea what some of it does so i just left it alone.

File: resources/domain.rb [hunk 1]
```diff
@@ -1,51 +1,69 @@
...
-      if myname.nil?
-        myname = "rule-#{seq}"
-        seq += 1
-      end
-      ulimit_rule "#{new_resource.name}:#{myname}" do
-        domain new_resource
-        instance_eval &code
-      end
-    end
+default_action :create
```
Comment (damacus) [inline resources/domain.rb]: This can go as the first action is the default action 👍

File: resources/domain.rb [hunk 2]
```diff
@@ -1,51 +1,69 @@
...
     end
   end
 
-  action :delete do
-    file ::File.join(node['ulimit']['security_limits_directory'], new_resource.filename) do
-      action :delete
-    end
+  new_resource.filename new_resource.name unless new_resource.filename
+  new_resource.filename "#{new_resource.filename}.conf"
+  template ::File.join(node['ulimit']['security_limits_directory'], new_resource.filename) do
```
Comment (damacus) [inline resources/domain.rb]: I've been converting these kinds of node attributes to properties else where.

Comment (tas50) [reply resources/domain.rb]: Yep we should get a property for this with a sane default

File: resources/rule.rb [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
-  property :item, kind_of: [Symbol, String], required: true
-  property :value, kind_of: [String, Numeric], required: true
-  property :domain, kind_of: [Chef::Resource, String], required: true
+load_current_value do
+  new_resource.domain new_resource.domain.domain_name if new_resource.domain.is_a?(Chef::Resource)
+  node.run_state[:ulimit] ||= Mash.new
+  node.run_state[:ulimit][new_resource.domain] ||= Mash.new
+end
+
+default_action :create
```
Comment (tas50) [inline resources/rule.rb]: This can go away since the first action is the default

**Pull request merged**